homeassistant/components/sensor/viaggiatreno.py

```diff
@@ -0,0 +1,190 @@
+    vol.Optional(CONF_STATION_NAME): cv.string,
+    })
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config,
+                         add_devices, discovery_info=None):
+    """Setup the ViaggiaTreno platform."""
+    train_id = config.get(CONF_TRAIN_ID)
+    station_id = config.get(CONF_STATION_ID)
+    name = config.get(CONF_NAME)
+    add_devices([ViaggiaTrenoSensor(hass, train_id, station_id, name)])
+
+
+@asyncio.coroutine
+def async_http_request(hass, uri):
+    """Perform actual request."""
+    try:
+        session = hass.helpers.aiohttp_client.async_get_clientsession(hass)
+        with async_timeout.timeout(REQUEST_TIMEOUT, loop=hass.loop):
+            req = yield from session.get(uri)
+        if req.status != 200:
+            return {'error': req.status}
+        else:
+            json_response = yield from req.json()
+            return json_response
+    except (asyncio.TimeoutError, aiohttp.ClientError) as exc:
+        _LOGGER.error("Cannot connect to ViaggiaTreno API endpoint: %s", exc)
+    except ValueError:
+        _LOGGER.error("Received non-JSON data from ViaggiaTreno API endpoint")
+
+
+class ViaggiaTrenoSensor(Entity):
+    """Implementation of a ViaggiaTreno sensor."""
+
+    def __init__(self, hass, train_id, station_id, name):
+        """Initialize the sensor."""
+        self.hass = hass
+        self._state = None
+        self._attributes = {}
+        self._unit = ''
+        self._icon = ICON
+        self._station_id = station_id
+
+        self.uri = VIAGGIATRENO_ENDPOINT.format(
+            station_id=station_id,
+            train_id=train_id)
+
+        if not name:
+            self._name = "Train {} from {}".format(train_id, station_id)
+        else:
+            self._name = name.format(train_id)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def device_state_attributes(self):
+        """Return extra attributes."""
+        self._attributes[ATTR_ATTRIBUTION] = CONF_ATTRIBUTION
+        return self._attributes
+
+    @staticmethod
+    def has_departed(data):
+        """Check if the train has actually departed."""
+        try:
+            first_station = data['fermate'][0]
+            if data['oraUltimoRilevamento'] or first_station['effettiva']:
+                return True
+        except ValueError:
+            _LOGGER.error('Cannot fetch first station: %s', data)
+        return False
+
+    @staticmethod
+    def has_arrived(data):
+        """Check if the train has already arrived."""
+        last_station = data["fermate"][-1]
```

Comment:
Please use quotes consistently. Short single word strings should be quoted with single quotes, if following the home assistant style.
https://home-assistant.io/developers/development_guidelines/#quotes